cmake/Modules/FindYAML.cmake

```diff
@@ -0,0 +1,30 @@
+# - Find libyaml
```

Comment:
libyaml installs it own CMake Config files, so do we need this?

Reply:
this is safer and works on all platforms we are running tests on. it is only used for the test environment. it doesn't have to be consistent with anything else.

Reply:
Good point, ubuntu has deloyed the cmake file yet: https://packages.ubuntu.com/eoan/amd64/libyaml-dev/filelist

We can remove this in next cmake refactor in a couple of years ;-)